homeassistant/components/simplisafe/config_flow.py

```diff
@@ -115,32 +75,139 @@ async def async_step_user(
             errors = {"base": "unknown"}
 
         if errors:
-            return self._async_show_form(errors=errors)
+            return self.async_show_form(
+                step_id=error_step_id,
+                data_schema=error_schema,
+                errors=errors,
+                description_placeholders={CONF_USERNAME: self._username},
+            )
+
+        assert self._simplisafe
+
+        if self._simplisafe.auth_state == AuthStates.PENDING_2FA_SMS:
+            return await self.async_step_sms_2fa()
+
+        try:
+            async with async_timeout.timeout(DEFAULT_EMAIL_2FA_TIMEOUT):
```

Comment:
IS 30 seconds enough to complete a 2FA via email?

Reply:
Hard to tell. Recommendation?

Reply:
5 minutes?

Noticing you need to check your email, maybe some bad luck: Need to log in. Email not arrived yet (delays happen). 30 seconds is a small window :)

Reply:
Got it. Thanks! 👍🏻